arbnode/sequencer.go

```diff
@@ -362,6 +364,31 @@ func (s *Sequencer) preTxFilter(_ *params.ChainConfig, header *types.Header, sta
 			return err
 		}
 	}
+	if options != nil && len(options.KnownAccounts) > 0 {
+		for address, rootHashOrSlots := range options.KnownAccounts {
+			if rootHashOrSlots.RootHash != nil {
+				trie := statedb.StorageTrie(address)
+				if trie == nil {
+					// TODO
+					return errors.New("Trie not found for address")
+				}
+				if trie.Hash() != *rootHashOrSlots.RootHash {
+					// TODO
+					return errors.New("Rejected")
+				}
+			} else if len(rootHashOrSlots.SlotValue) > 0 {
+				for slot, value := range rootHashOrSlots.SlotValue {
+					if statedb.GetState(address, slot) != value {
```

Comment:
I'd use `statedb.StorageTrie(address)` first so we can save that lookup and then do `trie.TryGet(slot.Bytes())` to get the value

Reply:
makes sense, fixed it